packages/application-shell-connectors/src/components/application-context/application-context.tsx

```diff
@@ -0,0 +1,217 @@
+import React from 'react';
+import moment from 'moment-timezone';
+import getDisplayName from '../../utils/get-display-name';
+
+type AdditionalProperties = { [key: string]: unknown };
+type TRawUser = {
+  id: string;
+  email: string;
+  firstName: string;
+  lastName: string;
+  language: string;
+  timeZone?: string;
+} & AdditionalProperties;
+type TApplicationContextUser = {
+  id: string;
+  email: string;
+  firstName: string;
+  lastName: string;
+  locale: string;
+  timeZone?: string;
+};
+type TApplicationContextPermissions = { [key: string]: boolean };
+type TRawProject = {
+  key: string;
+  version: number;
+  name: string;
+  countries: string[];
+  currencies: string[];
+  languages: string[];
+  owner: {
+    id: string;
+  };
+  permissions: TApplicationContextPermissions;
+} & AdditionalProperties;
+type TApplicationContextProject = {
+  key: string;
+  version: number;
+  name: string;
+  countries: string[];
+  currencies: string[];
+  languages: string[];
+  ownerId: string;
+};
+type TApplicationContextEnvironment = {
+  applicationName?: string;
+  frontendHost: string;
+  mcApiUrl: string;
+  location: string;
+  env: string;
+  cdnUrl: string;
+  servedByProxy: string | boolean;
+};
+type TApplicationContext<ExtendedEnvironment extends {}> = {
```

Comment:
You'll see a lot of generics `ExtendedEnvironment extends {}`.
The idea is that you can specify the additional environment props that are specific to your application, which are then applied to the `context.environment`.

At least I hope this is the correct way to do that.

Reply:
PS: we can also name it something like `AdditionalEnvironmentProperties` if it makes it clearer

Reply:
At gut feeling I'd fine `AdditionalEnvironmentProperties` easier to grasp but up to you. It's really picky.